orchard/orchard_runner.go

```diff
@@ -8,16 +8,52 @@ package orchard
 import (
 	"encoding/json"
 	"fmt"
+	"log"
 	"os/exec"
+	"strings"
+
+	"mce.salesforce.com/sprinkler/common"
 )
 
 type OrchardRunner interface {
-	Generate(artifact string, command string) string
+	Generate(artifact string, command string) (string, error)
 }
 
 type OrchardStdoutRunner struct{}
 
-func (self OrchardStdoutRunner) Generate(artifact string, command string) (string, error) {
+func (r OrchardStdoutRunner) Generate(artifact string, command string) (string, error) {
+	if artifact == "" {
+		return processCmd(command)
+	}
+
+	if !strings.HasPrefix(artifact, "s3://") {
+		return "", fmt.Errorf("artifact %v is not supported\n", artifact)
+	}
+
+	// download s3 artifact as local file
+	s3c, err := common.DefaultS3Client()
+	bb := common.S3Basics{S3Client: s3c}
+	s3bucketPath, err := bb.GetBucketPath(artifact)
+	localFile, err := bb.GetLastSegment(s3bucketPath.Path)
+	err = bb.DownloadFile(s3bucketPath.Bucket, s3bucketPath.Path, localFile)
```

Comment:
`err` is reassigned multiple times without being used in the middle, is this what we wanted?

Reply:
also you may want to make sure the `localFile` is stored in its own tmp directory so that it does not corrupt other command that uses file with the same name.

Reply:
I thought about this.  The client issued command need to be aware of the tmp directory as well, so that the command can get to it.  The service putting the workflow's localFile without the client's payload command key value being aware results in error.  

Will look into the err re-assignment and handling.

Reply:
refactored with handling for intermediate err.  Thank you for noting them.  

also the local file download includes the entire path from the s3 artifact s3key, to reduce chance of local file name collision.  The client's payload command needs to take into account / assumption that a local file be downloaded following the directory path as the S3 key /path.

Reply:
Thanks for the input.  refactored in new commit.